src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssetsIntegrationTest.cs

```diff
@@ -821,5 +823,398 @@ public void Build_Fails_WhenConflictingAssetsFoundBetweenAStaticWebAssetAndAFile
+                    var element = new XElement("StaticWebAsset", new XAttribute("Include", @"bundle\js\pkg-direct-dep.js"));
+                    element.Add(new XElement("SourceType"));
+                    element.Add(new XElement("SourceId", "PackageLibraryDirectDependency"));
+                    element.Add(new XElement("ContentRoot", "$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)\\bundle\\'))"));
+                    element.Add(new XElement("BasePath", "_content/PackageLibraryDirectDependency"));
+                    element.Add(new XElement("RelativePath", "js/pkg-direct-dep.js"));
+                    itemGroup.Add(element);
+                    project.Root.Add(itemGroup);
+                });
+
+            Directory.CreateDirectory(Path.Combine(projectDirectory.Path, "bundle", "js"));
+            File.WriteAllText(Path.Combine(projectDirectory.Path, "bundle", "js", "pkg-direct-dep.js"), "console.log('bundle');");
+
+            var pack = new MSBuildCommand(Log, "Pack", projectDirectory.Path, "PackageLibraryDirectDependency");
+            pack.Execute().Should().Fail();
+        }
+
+        // If you modify this test, make sure you also modify the test below this one to assert that things are not included as content.
+        [Fact]
+        public void Pack_IncludesStaticWebAssets()
+        {
+            var testAsset = "PackageLibraryDirectDependency";
+            var projectDirectory = CreateAspNetSdkTestAsset(testAsset, subdirectory: "TestPackages");
+
+            var pack = new MSBuildCommand(Log, "Pack", projectDirectory.Path, "PackageLibraryDirectDependency");
+            pack.WithWorkingDirectory(projectDirectory.Path);
+            var result = pack.Execute("/bl");
+
+            result.Should().Pass();
+
+            var outputPath = pack.GetOutputDirectory(DefaultTfm, "Debug").ToString();
+
+            new FileInfo(Path.Combine(outputPath, "PackageLibraryDirectDependency.dll")).Should().Exist();
+
+            result.Should().NuPkgContain(
+                Path.Combine(projectDirectory.Path, "PackageLibraryDirectDependency", "bin", "Debug", "PackageLibraryDirectDependency.1.0.0.nupkg"),
+                filePaths: new[]
+                {
+                    Path.Combine("staticwebassets", "js", "pkg-direct-dep.js"),
+                    Path.Combine("staticwebassets", "css", "site.css"),
+                    Path.Combine("staticwebassets", "PackageLibraryDirectDependency.bundle.scp.css"),
+                    Path.Combine("build", "Microsoft.AspNetCore.StaticWebAssets.props"),
+                    Path.Combine("build", "PackageLibraryDirectDependency.props"),
+                    Path.Combine("buildMultiTargeting", "PackageLibraryDirectDependency.props"),
+                    Path.Combine("buildTransitive", "PackageLibraryDirectDependency.props")
+                });
+        }
+
+        [Fact]
+        public void Pack_StaticWebAssets_WithoutFileExtension_AreCorrectlyPacked()
```

Comment:
https://github.com/dotnet/aspnetcore/issues/29561

GeneratePackageOnBuild doesn't matter (as the issue mentions)